src/PooledArrays.jl

```diff
@@ -377,12 +511,22 @@ function unsafe_pool_push!(pa::PooledArray{T,R}, val) where {T,R}
            )))
     end
     pool_idx = convert(R, _pool_idx)
+    if pa.refcount[] > 1
+        pa.invpool = copy(pa.invpool)
+        pa.pool = copy(pa.pool)
+        Threads.atomic_sub!(pa.refcount, 1)
+        pa.refcount = Threads.Atomic{Int}(1)
+    end
     pa.invpool[val] = pool_idx
     push!(pa.pool, val)
     pool_idx
 end
 
-Base.@propagate_inbounds function Base.setindex!(x::PooledArray, val, ind::Integer)
+# assume PooledArray is only used with Arrays as this is what _label does
+# this simplifies code below
+Base.IndexStyle(::Type{<:PooledArray}) = IndexLinear()
```

Comment:
This is breaking but without it tests in e.g. https://github.com/JuliaData/PooledArrays.jl/pull/56/files#diff-3b9314a6f9f2d7eec1d0ef69fa76cfabafdbe6d0df923768f9ec32f27a249c63R409 fail

I would let us bump to 1.2 though.

Reply:
We could also just return whatever `IndexStyle(A.refs)` returns, right? But is it really breaking? As long as we define the relevant methods, the cartesian index will be computed automatically when indexing into `A.refs` (it will just be slower, but maybe it's not so relevant here given that a dict lookup is involved).

Reply:
> We could also just return whatever IndexStyle(A.refs) returns, right?

I would leave it as I propose. The reason is that `IndexLinear` and `IndexCartesian` require different methods to be implemented as https://docs.julialang.org/en/v1/manual/interfaces/#man-interface-array specifies.

I opt for defining methods for `IndexLinear` as this is simpler to maintain and probably nobody passes e.g. sparse array to `PooledArray` anyway.

This is only breaking because `IndexStyle` returns something else, so not a problem. I would leave things as I proposed.